game/src/ecs/systems/HealthSystem.java

```diff
@@ -29,7 +29,15 @@ public void update() {
                 // Apply damage
                 .map(this::applyDamage)
                 // Filter all dead entities
-                .filter(hsd -> hsd.hc.getCurrentHealthpoints() <= 0)
+                .filter(hsd -> hsd.hc.isDead())
+                .filter(
+                        hsd -> {
+                            if (hsd.hc.getDeathAnimation() == null) return true;
+                            if (!hsd.ac.getCurrentAnimation().equals(hsd.hc.getDeathAnimation())) {
+                                hsd.ac.setCurrentAnimation(hsd.hc.getDeathAnimation());
+                            }
+                            return hsd.ac.getCurrentAnimation().isFinished();
+                        })
```

Comment:
Also geht es hier eigentlich nur um diesen Code schnipsel, oder? Sieht für mich erstmal gut aus. Was passiert, wenn ich als Todes Animation eine nicht endende Animation ausgewählt habe?

Reply:
You will die Forever

Reply:
Nö

Reply:
Hää, wer will denn nicht ewig sterben 🤪
Sollte fixed sein.